ethergo/submitter/chain_queue.go

```diff
@@ -85,30 +92,41 @@ func (t *txSubmitterImpl) chainPendingQueue(parentCtx context.Context, chainID *
 
 		cq.bumpTX(gCtx, tx)
 	}
-    cq.updateOldTxStatuses(gCtx)
+	cq.updateOldTxStatuses(gCtx)
 
 	err = cq.g.Wait()
 	if err != nil {
 		return fmt.Errorf("error in chainPendingQueue: %w", err)
 	}
 
-	sort.Slice(cq.reprocessQueue, func(i, j int) bool {
-		return cq.reprocessQueue[i].Nonce() < cq.reprocessQueue[j].Nonce()
-	})
-
-	calls := make([]w3types.Caller, len(cq.reprocessQueue))
-	txHashes := make([]common.Hash, len(cq.reprocessQueue))
-	for i, tx := range cq.reprocessQueue {
-		calls[i] = eth.SendTx(tx.Transaction).Returns(&txHashes[i])
-	}
-
-	cq.storeAndSubmit(ctx, calls, span)
+	cq.storeAndSubmit(ctx, cq.reprocessQueue, span)
 
 	return nil
 }
 
 // storeAndSubmit stores the txes in the database and submits them to the chain.
-func (c *chainQueue) storeAndSubmit(ctx context.Context, calls []w3types.Caller, span trace.Span) {
+func (c *chainQueue) storeAndSubmit(ctx context.Context, reprocessQueue []db.TX, parentSpan trace.Span) {
```

Comment:
Do we need the `reprocessQueue` arg here if we already have access to `c.reprocesQueue`?